sotodlib/preprocess/processes.py

```diff
@@ -1212,66 +1211,112 @@ class SourceFlags(_Preprocess):
                                                           overwrite=self.calc_cfgs.get('overwrite', True),
                                                           source_flags_name=self.calc_cfgs.get('source_flags_name', None),
                                                           mask=self.calc_cfgs.get('mask', None),
-                                                          center_on=p[0],
+                                                          center_on=center_on,
                                                           res=self.calc_cfgs.get('res', None),
                                                           max_pix=self.calc_cfgs.get('max_pix', None))
 
-            source_aman.wrap(p[0], source_flags, [(0, 'dets'), (1, 'samps')])
+            source_aman.wrap(source, source_flags, [(0, 'dets'), (1, 'samps')])
+
+            if self.calc_cfgs.get('inv_flag') is not None:
+                if self.calc_cfgs['inv_flag']:    
+                    source_aman.wrap(source+ '_inv',
+                                    RangesMatrix.from_mask(~source_flags.mask()),
+                                    [(0, 'dets'), (1, 'samps')])
 
         # add sources that were not nearby from source list
         for source in source_list:
+            if isinstance(source, tuple):
+                source = source[0]
             if source not in source_aman._fields:
                 source_aman.wrap(source, RangesMatrix.zeros([aman.dets.count, aman.samps.count]),
                                  [(0, 'dets'), (1, 'samps')])
+                
+                if self.calc_cfgs.get('inv_flag'):
+                    source_aman.wrap(source + '_inv',
+                                    RangesMatrix.ones([aman.dets.count, aman.samps.count]),
+                                    [(0, 'dets'), (1, 'samps')])
 
         self.save(proc_aman, source_aman)
 
     def save(self, proc_aman, source_aman):
         if self.save_cfgs is None:
             return
         if self.save_cfgs:
-            proc_aman.wrap("source_flags", source_aman)
+            if self.calc_cfgs.get('source_flags_name') is not None:
+                proc_aman.wrap(self.calc_cfgs.get('source_flags_name', None), source_aman)
+            else:
+                proc_aman.wrap("source_flags", source_aman)
 
     def select(self, meta, proc_aman=None):
         if self.select_cfgs is None:
             return meta
         if proc_aman is None:
             source_flags = meta.preprocess.source_flags
         else:
-            source_flags = proc_aman.source_flags
+            if self.calc_cfgs.get('source_flags_name') is not None:
+                source_flags = proc_aman[self.calc_cfgs.get('source_flags_name')]
+            else:
+                source_flags = proc_aman.source_flags
 
-        source_list = np.atleast_1d(self.calc_cfgs.get('center_on', 'planet'))
-        if source_list == ['planet']:
+        select_list = np.atleast_1d(self.select_cfgs["select_source"])
```

Comment:
Give an example of this in the example config entry(ies) in the docstring. Currently I think if you passed in `self.select_cfgs = True` as is done in the current docstring example this will fail bc you're missing `select_source` for that reason to observe backwards compatibility choose a reasonable default--probably all source flags.